tests/components/vera/test_init.py

```diff
@@ -42,6 +45,73 @@
     assert entry1
 
 
+async def test_init_from_file(
+    hass: HomeAssistant, vera_component_factory: ComponentFactory
+) -> None:
+    """Test function."""
+    vera_device1 = MagicMock(spec=pv.VeraBinarySensor)  # type: pv.VeraBinarySensor
+    vera_device1.device_id = 1
+    vera_device1.vera_device_id = 1
+    vera_device1.name = "first_dev"
+    vera_device1.is_tripped = False
+    entity1_id = "binary_sensor.first_dev_1"
+
+    await vera_component_factory.configure_component(
+        hass=hass,
+        controller_config=new_simple_controller_config(
+            config={CONF_CONTROLLER: "http://127.0.0.1:111"},
+            config_from_file=True,
+            serial_number="first_serial",
+            devices=(vera_device1,),
+        ),
+    )
+
+    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+    entry1 = entity_registry.async_get(entity1_id)
+    assert entry1
+
+
+async def test_async_setup_remove_configs(hass: HomeAssistant) -> None:
+    """Test function."""
+    entry1 = MagicMock(spec=ConfigEntry)
+    entry1.entry_id = "id1"
+    entry1.domain = DOMAIN
+    entry1.data = {CONF_CONTROLLER: "url1", CONF_SOURCE: config_entries.SOURCE_IMPORT}
+
+    entry2 = MagicMock(spec=ConfigEntry)
+    entry2.entry_id = "id2"
+    entry2.domain = DOMAIN
+    entry2.data = {CONF_CONTROLLER: "url2", CONF_SOURCE: config_entries.SOURCE_USER}
+
+    hass.config_entries.async_entries = MagicMock(return_value=[entry1, entry2])
```

Comment:
Don't patch the core. Keep it intact and check that the correct entry data is present after updating the entries.

Reply:
I'm mocking core because I don't see a good way to add config entries before running the setup hook.

Reply:
Use this:
https://github.com/home-assistant/home-assistant/blob/7d506bc38b73f5a81649a0fe629f054ea5dbdae6/tests/common.py#L692-L694